internal/otelcollector/config/log/agent/config_builder.go

```diff
@@ -30,7 +30,7 @@ type BuildOptions struct {
 	AgentNamespace              string
 	ClusterName                 string
 	CloudProvider               string
-	Enrichments                 processors.Enrichments
+	Enrichments                 *operatorv1alpha1.EnrichmentSpec
```

Comment:
Maybe let's pass a value instead of a pointer to avoid nil checks?

Reply:
This field is optional